### PR TITLE
feat: adds org-level compliance setup

### PR DESCRIPTION
**Pull request by sameer-in**
- Adding new example for GCP Org using workload identity
- we only deploy the artifacts in current project
- all other projects uses current project as workload identity

### CODE REVIEW

File: examples/org-workload-identity-provider/versions.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      configuration_aliases = [google.multiproject]
+    }
+    google-beta = {
+      source                = "hashicorp/google-beta"
+      version               = ">= 4.21.0"
+      configuration_aliases = [google-beta.multiproject]
+    }
+    sysdig = {
+      source  = "local/sysdiglabs/sysdig"
+      version = "~> 1.0.0"
```
Comment (wideawakening) [inline examples/org-workload-identity-provider/versions.tf:17]: careful with these.
i guess you were testing local provider changes, so let's first publish those changes in the provider repo

**Comment by wideawakening** [comment]
some note son a quick review

- since this is a public repo  internal jira tickets does not give context so adding a quick description would help :)
from my context, this example aims to replace current organizational approach for organizational setup, creating a single workload identity on the org-level, instead of the current projet-level.
   - dependant on [sysdig tf provider changes](https://github.com/sysdiglabs/terraform-provider-sysdig/pull/222)
- let's try to tidy up submodules
 - let's try to merge both cloudbench submodules into a single one, unless there's a reason not to do so
  - currently existing `/task` submodule should be used, instead of duplicated, as there is no change with the current one
  - current `trust_relationship` should be refactored, but quick looking at the changes think it's ok to split them into `trust_relations_single_project` and `trust_relationship_organizational`
  - let's replace current `/examples/organization` since this approach is more optimized, don't  see a reason to maintain previous.

File: examples/org-workload-identity-provider/versions.tf (hunk 1 above)
Comment (sameer-in) [reply examples/org-workload-identity-provider/versions.tf:17]: Yeah. I added it for testing it locally. I will remove it once we have a release.

File: examples/organization/README.md [hunk 2]
```diff
@@ -88,7 +88,8 @@ module "secure-for-cloud_example_organization" {
   }
 
   source = "sysdiglabs/secure-for-cloud/google//examples/organization"
-  organization_domain       = "<ORG_DOMAIN>"
+  organization_domain = "<ORG_DOMAIN>"
+  name                = "<NAME>"
```
Comment (wideawakening) [inline examples/organization/README.md:92]: in docs we only put mandatory attributes, or if optional, state it's optional to avoid confussion

Comment (sameer-in) [reply examples/organization/README.md]: Removed this.

**Comment by sameer-in** [comment]
@wideawakening Created a separate example. It has a different cloud-bench module.

**Review by tembleking — APPROVED** [review]

**Review by wideawakening — APPROVED** [review]

**Comment by wideawakening** [comment]
ahhh finally! so many non-deterministic problems with the linting/validation....  🥳 

@sameer-in  i changed the following topics we agreed while peering
- modified the organizational diagrams to reflect that it does not only deploy a role, but a workload identity pool + role + sa
- renamed the new example to `organization-org_compliance` to reflect its the same as the `organization` one plus the optimization for compliance
- also modified the snipped in the example since it was pointing the old one
- added the deprecated message on the `organizational`
- tried to add a new test for this one, to be covered by GH actions, but since we cannot re-spawn it over and over (due to the softdelete) y removed that check. i left the test itself for local testin

**Pull request merged**